### DISCUSSION TITLE
Setting IP addresses of iCubLisboa01 PC and control boards for our internal network

### DISCUSSION THREAD

**Discussion by plinioMoreno** [robotology/community#601] [Q&A] [answered]
Dear all,

We received our upgraded iCub, thanks to all the persons involved in this!

Our iCub was previously configured on our local network [10.10.1.40- 10.10.1.50](http://mediawiki.isr.ist.utl.pt/wiki/ICub_machines_configuration). We have three additional servers that were configured to run the iCub software.

We want to configure our upgraded iCub on this network, so my question is: How can we change the IP addresses of the controlboards and the icub-head pc to our 10.10.1.X network? Is this possible to change all these IPs without unplugging anything and running FirmwareUpdater? About the first question, we got an initial idea on this as follows:

From the file [network.iCubLisboa01.xml](https://github.com/robotology/robots-configuration/blob/59add72028f640bfdcc95074f9c613b34fcbc1e3/iCubLisboa01/network.iCubLisboa01.xml) we guessed that all the ETH boards are connected in a single series of boards. If that is correct, I would say that we should start the IP address change from the last board in the chain (10.0.1.9) and then change the IP address of the board 10.0.1.11, then 10.0.1.8 and so on. Please let me know if this is correct.

If we can do this change with the FirmwareUpdater, please provide us with pointers on how to compile it on the installed software on the icub-head. 

Thanks in advance.

**Comment by pattacini** [comment, marked as answer]
Hi @plinioMoreno 

The `icub-head` has 2 ETH networks:
- The internal ETH 10.0.1.x to communicate with the embedded boards. See https://icub-tech-iit.github.io/documentation/icub_operating_systems/icubos/networking/?h=wifi#motor-network-configurations.
- The external ETH 10.0.0.x to communicate with the outside. See https://icub-tech-iit.github.io/documentation/icub_operating_systems/icubos/networking/?h=wifi#external-network-configurations.

So, to achieve your goal, you need to fiddle with the external ETH configuration ONLY, inside the `icub-head`.
Therefore, you won't be connecting your servers directly to the internal network. That said, the `icub-head` will be streaming the data to/from the embedded boards onto the external network.

cc @davidelasagna

**Comment by plinioMoreno** [comment]
Hi @pattacini,

Thanks for the pointers to the documentation. Then we will go ahead and change the external ETH IP address.
